Tools/autotest/common.py

```diff
@@ -1624,8 +1624,11 @@ def check_test_syntax(self, test_file):
                 # check for lambda: test_function without paranthesis
                 faulty_strings = re.findall(r"lambda\s*:\s*\w+.\w+\s*\)", f.read())
                 if faulty_strings:
-                    desc = ("Syntax error in autotest lambda at : " +
-                            faulty_strings)
+                    desc = "Syntax error in autotest lambda at :"
+                    print(desc)
+                    for x in range(len(faulty_strings)):
+                        desc = faulty_strings[x]
+                        print(desc)
```

Comment:
I agree that we should print the faulty strings, but this now seems a weird way to do things: we print an error message, then throw an exception with that message in details, then catch the exception and print a separate message.

By the way, you are passing only the last faulty string to the Exception, that's clearly a bug.

Reply:
right, corrected